File: step_definitions/Usability Testing for Ease of Navigation.js
import { Given, When, Then } from "cypress-cucumber-preprocessor/steps";
import NavigationPage from "../pages/navigationPage";

Given("User A is on the login page", () => {
 NavigationPage.visitLoginPage();
});

When("User A logs in", () => {
 NavigationPage.loginAsUserA();
});

Then("User A should see the home page with navigation options", () => {
 NavigationPage.verifyHomePage();
});

Given("User A is on the home page", () => {
 NavigationPage.verifyHomePage();
});

When("User A navigates to the product catalog", () => {
 NavigationPage.navigateToProductCatalog();
});

Then("User A should access the product catalog without any issues", () => {
 NavigationPage.verifyProductCatalogPage();
});

Given("User A is on the product catalog page", () => {
 NavigationPage.verifyProductCatalogPage();
});

When("User A attempts to access account settings", () => {
 NavigationPage.accessAccountSettings();
});

Then("User A should successfully navigate to account settings", () => {
 NavigationPage.verifyAccountSettingsPage();
});

Given("User A is logged in", () => {
 NavigationPage.verifyUserALoggedIn();
});

When("User A logs out and User B logs in", () => {
 NavigationPage.logoutAndLoginAsUserB();
});

Then("User B should log in successfully", () => {
 NavigationPage.verifyUserBLoggedIn();
});

Given("User B is on the home page", () => {
 NavigationPage.verifyHomePage();
});

When("User B accesses the admin dashboard", () => {
 NavigationPage.accessAdminDashboard();
});

Then("User B should access the admin dashboard seamlessly", () => {
 NavigationPage.verifyAdminDashboardPage();
});

Given("User B is on the admin dashboard", () => {
 NavigationPage.verifyAdminDashboardPage();
});

When("User B navigates to user management", () => {
 NavigationPage.navigateToUserManagement();
});

Then("User B should navigate to user management without difficulty", () => {
 NavigationPage.verifyUserManagementPage();
});

Given("User B is logged in", () => {
 NavigationPage.verifyUserBLoggedIn();
});

When("User B logs out and User C logs in", () => {
 NavigationPage.logoutAndLoginAsUserC();
});

Then("User C should log in successfully", () => {
 NavigationPage.verifyUserCLoggedIn();
});

Given("User C is logged in", () => {
 NavigationPage.verifyUserCLoggedIn();
});

When("User C attempts to navigate to the checkout page", () => {
 NavigationPage.navigateToCheckoutPage();
});

Then("User C should be redirected to the login page", () => {
 NavigationPage.verifyRedirectedToLoginPage();
});

When("User C views navigation options", () => {
 NavigationPage.viewNavigationOptions();
});

Then("User C should see limited navigation options appropriate for guest users", () => {
 NavigationPage.verifyLimitedNavigationOptions();
});

Given("User is on any page", () => {
 NavigationPage.verifyUserOnAnyPage();
});

When("User uses keyboard shortcuts for navigation", () => {
 NavigationPage.useKeyboardShortcuts();
});

Then("All keyboard shortcuts should work as intended", () => {
 NavigationPage.verifyKeyboardShortcuts();
});

Given("User is on the mobile interface", () => {
 NavigationPage.verifyMobileInterface();
});

When("User navigates through the application", () => {
 NavigationPage.navigateThroughMobileInterface();
});

Then("Navigation should be smooth and intuitive", () => {
 NavigationPage.verifySmoothNavigation();
});

When("User tries to access a non-existent page", () => {
 NavigationPage.accessNonExistentPage();
});

Then("User should receive an error message indicating the page does not exist", () => {
 NavigationPage.verifyErrorMessageForNonExistentPage();
});

Given("User is logged in with any role", () => {
 NavigationPage.verifyUserLoggedInWithAnyRole();
});

When("User views navigation options", () => {
 NavigationPage.viewNavigationOptions();
});

Then("Navigation options should be consistent and appropriate for each user role", () => {
 NavigationPage.verifyConsistentNavigationOptions();
});

Given("User is navigating through the application", () => {
 NavigationPage.verifyUserNavigatingThroughApplication();
});

When("User follows navigation paths", () => {
 NavigationPage.followNavigationPaths();
});

Then("All navigation paths should align with the expected flow diagrams", () => {
 NavigationPage.verifyNavigationPaths();
});

Given("Server is under high load conditions", () => {
 NavigationPage.verifyServerUnderHighLoad();
});

When("User navigates through the application", () => {
 NavigationPage.navigateThroughApplication();
});

Then("Navigation should remain responsive and functional", () => {
 NavigationPage.verifyResponsiveNavigation();
});

File: step_definitions/Usability Testing for Ease of Navigation.java
package com.cucumber.steps;

import com.page_objects.NavigationPage;
import io.cucumber.java.en.Given;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;

public class NavigationSteps extends NavigationPage {

    @Given("User A is on the login page")
    public void userAIsOnTheLoginPage() {
        navigateToLoginPage();
    }

    @When("User A logs in")
    public void userALogsIn() {
        loginAsUserA();
    }

    @Then("User A should see the home page with navigation options")
    public void userASeesHomePageWithNavigationOptions() {
        verifyHomePageNavigationOptions();
    }

    @Given("User A is on the home page")
    public void userAIsOnTheHomePage() {
        verifyUserAOnHomePage();
    }

    @When("User A navigates to the product catalog")
    public void userANavigatesToProductCatalog() {
        navigateToProductCatalog();
    }

    @Then("User A should access the product catalog without any issues")
    public void userAAccessesProductCatalog() {
        verifyProductCatalogAccess();
    }

    @Given("User A is on the product catalog page")
    public void userAIsOnTheProductCatalogPage() {
        verifyUserAOnProductCatalogPage();
    }

    @When("User A attempts to access account settings")
    public void userAAttemptsToAccessAccountSettings() {
        navigateToAccountSettings();
    }

    @Then("User A should successfully navigate to account settings")
    public void userANavigatesToAccountSettings() {
        verifyAccountSettingsAccess();
    }

    @Given("User A is logged in")
    public void userAIsLoggedIn() {
        verifyUserALoggedIn();
    }

    @When("User A logs out and User B logs in")
    public void userALogsOutAndUserBLogsIn() {
        logoutAndLoginAsUserB();
    }

    @Then("User B should log in successfully")
    public void userBLogsInSuccessfully() {
        verifyUserBLoginSuccess();
    }

    @Given("User B is on the home page")
    public void userBIsOnTheHomePage() {
        verifyUserBOnHomePage();
    }

    @When("User B accesses the admin dashboard")
    public void userBAccessesAdminDashboard() {
        navigateToAdminDashboard();
    }

    @Then("User B should access the admin dashboard seamlessly")
    public void userBAccessesAdminDashboardSeamlessly() {
        verifyAdminDashboardAccess();
    }

    @Given("User B is on the admin dashboard")
    public void userBIsOnTheAdminDashboard() {
        verifyUserBOnAdminDashboard();
    }

    @When("User B navigates to user management")
    public void userBNavigatesToUserManagement() {
        navigateToUserManagement();
    }

    @Then("User B should navigate to user management without difficulty")
    public void userBNavigatesToUserManagement() {
        verifyUserManagementAccess();
    }

    @Given("User B is logged in")
    public void userBIsLoggedIn() {
        verifyUserBLoggedIn();
    }

    @When("User B logs out and User C logs in")
    public void userBLogsOutAndUserCLogsIn() {
        logoutAndLoginAsUserC();
    }

    @Then("User C should log in successfully")
    public void userCLogsInSuccessfully() {
        verifyUserCLoginSuccess();
    }

    @Given("User C is logged in")
    public void userCIsLoggedIn() {
        verifyUserCLoggedIn();
    }

    @When("User C attempts to navigate to the checkout page")
    public void userCAttemptsToNavigateToCheckoutPage() {
        navigateToCheckoutPage();
    }

    @Then("User C should be redirected to the login page")
    public void userCIsRedirectedToLoginPage() {
        verifyRedirectToLoginPage();
    }

    @When("User C views navigation options")
    public void userCViewsNavigationOptions() {
        viewNavigationOptions();
    }

    @Then("User C should see limited navigation options appropriate for guest users")
    public void userCSeesLimitedNavigationOptions() {
        verifyLimitedNavigationOptions();
    }

    @Given("User is on any page")
    public void userIsOnAnyPage() {
        verifyUserOnAnyPage();
    }

    @When("User uses keyboard shortcuts for navigation")
    public void userUsesKeyboardShortcuts() {
        useKeyboardShortcuts();
    }

    @Then("All keyboard shortcuts should work as intended")
    public void keyboardShortcutsWorkAsIntended() {
        verifyKeyboardShortcutsFunctionality();
    }

    @Given("User is on the mobile interface")
    public void userIsOnMobileInterface() {
        verifyUserOnMobileInterface();
    }

    @When("User navigates through the application")
    public void userNavigatesThroughApplication() {
        navigateThroughApplication();
    }

    @Then("Navigation should be smooth and intuitive")
    public void navigationIsSmoothAndIntuitive() {
        verifySmoothNavigation();
    }

    @When("User tries to access a non-existent page")
    public void userTriesToAccessNonExistentPage() {
        accessNonExistentPage();
    }

    @Then("User should receive an error message indicating the page does not exist")
    public void userReceivesErrorMessage() {
        verifyErrorMessageForNonExistentPage();
    }

    @Given("User is logged in with any role")
    public void userIsLoggedInWithAnyRole() {
        verifyUserLoggedInWithAnyRole();
    }

    @When("User views navigation options")
    public void userViewsNavigationOptions() {
        viewNavigationOptions();
    }

    @Then("Navigation options should be consistent and appropriate for each user role")
    public void navigationOptionsAreConsistent() {
        verifyConsistentNavigationOptions();
    }

    @Given("User is navigating through the application")
    public void userIsNavigatingThroughApplication() {
        verifyUserNavigatingThroughApplication();
    }

    @When("User follows navigation paths")
    public void userFollowsNavigationPaths() {
        followNavigationPaths();
    }

    @Then("All navigation paths should align with the expected flow diagrams")
    public void navigationPathsAlignWithFlowDiagrams() {
        verifyNavigationPathsAlignment();
    }

    @Given("Server is under high load conditions")
    public void serverIsUnderHighLoadConditions() {
        simulateHighLoadConditions();
    }

    @When("User navigates through the application")
    public void userNavigatesThroughApplicationUnderLoad() {
        navigateThroughApplicationUnderLoad();
    }

    @Then("Navigation should remain responsive and functional")
    public void navigationRemainsResponsive() {
        verifyResponsiveNavigationUnderLoad();
    }
}